File: packages/sdk-typescript/test-basic-functionality.ts
/**
 * Basic functionality test for the TypeScript SDK
 */

import { createDevnetClient } from './src/client-v2';

async function testBasicFunctionality() {
  try {
    console.log('🔧 Testing TypeScript SDK basic functionality...');
    
    // Create a devnet client
    const client = createDevnetClient();
    console.log('✅ Client created successfully');
    
    // Test connection
    const isConnected = await client.isConnected();
    console.log('🌐 Connection status:', isConnected ? 'Connected' : 'Not connected');
    
    // Get cluster info
    try {
      const clusterInfo = await client.getClusterInfo();
      console.log('📊 Cluster info:', clusterInfo);
    } catch (error) {
      console.log('⚠️ Could not get cluster info (expected on devnet):', error instanceof Error ? error.message : String(error));
    }
    
    // Test RPC access
    const rpc = client.getRpc();
    console.log('🔗 RPC client available:', !!rpc);
    
    // Test program ID access
    const programId = client.getProgramId();
    console.log('🎯 Program ID:', programId);
    
    // Test services access
    const agents = client.agents;
    const channels = client.channels;
    console.log('🤖 Agent service available:', !!agents);
    console.log('📢 Channel service available:', !!channels);
    
    console.log('✅ All basic functionality tests passed!');
    return true;
  } catch (error) {
    console.error('❌ Basic functionality test failed:', error);
    return false;
  }
}

// Run the test
testBasicFunctionality()
  .then((success) => {
    process.exit(success ? 0 : 1);
  })
  .catch((error) => {
    console.error('Test execution failed:', error);
    process.exit(1);
  }); 

File: packages/sdk-rust/examples/test_basic_functionality.rs
/**
 * Basic functionality test for the Rust SDK
 */

use podai_sdk::client::*;
use podai_sdk::services::agent::AgentService;
use solana_sdk::pubkey::Pubkey;
use std::str::FromStr;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("🔧 Testing Rust SDK basic functionality...");
    
    // Create devnet RPC URL
    let rpc_url = "https://api.devnet.solana.com";
    
    // Test program ID parsing
    let program_id = match Pubkey::from_str("HEpGLgYsE1kP8aoYKyLFc3JVVrofS7T4zEA6fWBJsZps") {
        Ok(id) => {
            println!("✅ Program ID parsed successfully: {}", id);
            id
        }
        Err(e) => {
            println!("❌ Failed to parse program ID: {}", e);
            return Err(e.into());
        }
    };
    
    // Test agent service creation
    match AgentService::new(rpc_url, program_id) {
        Ok(agent_service) => {
            println!("✅ Agent service created successfully");
            
            // Test getting RPC client
            let _rpc_client = agent_service.get_rpc_client();
            println!("✅ RPC client accessible");
            
            println!("✅ All basic functionality tests passed!");
        }
        Err(e) => {
            println!("❌ Failed to create agent service: {}", e);
            return Err(e.into());
        }
    }
    
    Ok(())
} 